src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -449,7 +449,7 @@ private async Task<ImmutableArray<Diagnostic>> GetAllDiagnosticsWithoutStateTrac
                 driver.Initialize(compilation, _analysisOptions, compilationData, categorizeDiagnostics, cancellationToken);
                 var hasAllAnalyzers = analyzers.Length == Analyzers.Length;
                 var analysisScope = new AnalysisScope(compilation, _analysisOptions.Options, analyzers, hasAllAnalyzers, concurrentAnalysis: _analysisOptions.ConcurrentAnalysis, categorizeDiagnostics: categorizeDiagnostics);
-                driver.AttachQueueAndStartProcessingEvents(compilation.EventQueue, analysisScope, cancellationToken);
+                driver.AttachQueueAndStartProcessingEvents(compilation.EventQueue!, analysisScope, cancellationToken);
```

Comment:
💭  Is it possible to annotate `WithEventQueue` to inform flow analysis?

Reply:
I don't think we have a mechanism for that at the moment.

---
In reply to: [467152746](https://github.com/dotnet/roslyn/pull/46409#discussion_r467152746) [](ancestors = 467152746)

Reply:
Why not? I would think that `MemberNotNull` would work? Unless `WithEventQueue` can take a nullable parameter?

---
In reply to: [468144187](https://github.com/dotnet/roslyn/pull/46409#discussion_r468144187) [](ancestors = 468144187,467152746)

Reply:
I think `WithEventQueue` accepts a null input, so we can't just use `MemberNotNull`. And it doesn't return a boolean, so we can't use `MemberNotNullWhen`...

---
In reply to: [468899201](https://github.com/dotnet/roslyn/pull/46409#discussion_r468899201) [](ancestors = 468899201,468144187,467152746)